docs/recipes/upgrading/v4-to-v5.md

```diff
@@ -0,0 +1,274 @@
+
+	```diff
+	- "start": "webpack-dev-server --mode development",
+	- "build": "webpack --mode production",
+	- "test:live": "webpack-dev-server --mode development --env test"
+	```
+	
+1. Update the left over NPM dependencies:
+
+	- You can run to upgrade all at once `npx npm-check-updates -u` or if you prefer to be more conservative you can use `npx npm-check-updates -i`.
+	
+	If you use the second option, make sure to select `react`, `react-dom`, `sass`.
+	
+1. Install the new NPM dependencies
+
+	`npm install -D vite @vitejs/plugin-react tailwindcss postcss autoprefixer @types/node`
+	
+1. Delete `webpack.config.js`
+
+1. Create the file `src/Client/vite.config.mts` with the following content
+	
+	```ts
+	import { defineConfig } from "vite";
+	import react from "@vitejs/plugin-react";
+	
+	const proxyPort = process.env.SERVER_PROXY_PORT || "5000";
+	const proxyTarget = "http://localhost:" + proxyPort;
+	
+	// https://vitejs.dev/config/
+	export default defineConfig({
+	    plugins: [react()],
+	    build: {
+	        outDir: "../../deploy/public",
+	    },
+	    server: {
+	        port: 8080,
+	        proxy: {
+	            // redirect requests that start with /api/ to the server on port 5000
+	            "/api/": {
+	                target: proxyTarget,
+	                changeOrigin: true,
+	            }
+	        }
+	    }
+	});
+	```
+
+1. Overwrite `src/Client/index.html` with the following content
+	
+	```html
+	<!doctype html>
+	<html>
+	<head>
+	    <title>SAFE Template</title>
+	    <meta charset="utf-8">
+	    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+	    <link rel="icon" type="image/png" href="/favicon.png"/>
+	    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css">
+	</head>
+	<body>
+	    <div id="elmish-app"></div>
+	    <!-- Polyfill for remotedev -->
+	    <script type="text/javascript">
+	        var global = global || window;
+	    </script>
+	    <script type="module" src="/output/App.js"></script>
+	</body>
+	</html>
+	```
+	
+1. Run `npx tailwindcss init -p` in `src/Client`
+1. Change `tailwind.config.js` content with
+
+	```js
+	/** @type {import('tailwindcss').Config} */
+	module.exports = {
+	    mode: "jit",
+	    content: [
+	        "./index.html",
+	        "./**/*.{fs,js,ts,jsx,tsx}",
+	    ],
+	    theme: {
+	        extend: {},
+	    },
+	    plugins: []
+	}
+	```
+
+1. Create a file `src/Client/index.css` with the following content
+
+	```css
+	@tailwind base;
+	@tailwind components;
+	@tailwind utilities;
+	```
+	
+1. Add the following lines at the top of `src/Client/App.fs`
+	
+	```fs
+	open Fable.Core.JsInterop
+	
+	importSideEffects "./index.css"
+	```
+
+1. In `src/Client/Index.fs` replace all view code after the `update` function with the code [here](https://github.com/SAFE-Stack/SAFE-template/blob/master/Content/default/src/Client/Index.fs)
```

Comment:
This isn't very precise. It's probably better to use a permalink and refer to specific lines (e.g. https://github.com/SAFE-Stack/SAFE-template/blob/064b902c4ca8229f8f671a3fab67c2871cd772c1/Content/default/src/Client/Index.fs#L42-L112)